neo4j/resultsummary.go

```diff
@@ -586,9 +737,96 @@ func (n *notification) Position() InputPosition {
 func (n *notification) Offset() int {
 	return n.notification.Position.Offset
 }
+
 func (n *notification) Column() int {
 	return n.notification.Position.Column
 }
+
 func (n *notification) Line() int {
 	return n.notification.Position.Line
 }
+
+type gqlStatusObject struct {
+	gqlStatusObject *db.GqlStatusObject
+}
+
+func (g *gqlStatusObject) GqlStatus() string {
+	return g.gqlStatusObject.GqlStatus
+}
+
+func (g *gqlStatusObject) StatusDescription() string {
+	return g.gqlStatusObject.StatusDescription
+}
+
+func (g *gqlStatusObject) Position() InputPosition {
+	if g.gqlStatusObject.Position == nil {
+		return nil
+	}
+	return g
+}
```

Comment:
Is there any good reason to have this method, if all it does is return itself? This also means this `g.Offset()` et al. might blow up when called while `g.gqlStatusObject.Position == nil` or am I wrong?

Reply:
It returns itself which is an interface of InputPosition, so from the users code they will write:

```
for _, gqlStatusObject := range gqlStatusObjects {
    pos := gqlStatusObject.Position()
    // you can't do gqlStatusObject.Offset() as the interface doesn't allow it. You're expected to check if
    // gqlStatusObject.Position() is nil before accessing gqlStatusObject.Position().Line() etc.
}
```

It's the same design pattern as the Notification interface: https://github.com/neo4j/neo4j-go-driver/blob/2525e8f57093834960c55afd7519869df8954bfc/neo4j/resultsummary.go#L729-L746

Happy to discuss